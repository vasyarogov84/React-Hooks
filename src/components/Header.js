import React, { useContext, useState } from 'react';
import AuthContext from '../auth-context';

const Header = (props) => {
    const [name, changeName] = useState('Viktor');
    const auth = useContext(AuthContext);

    const changePersonName = () => {
        if (name === 'Viktor') {
            changeName('Iryna');
        } else {
            changeName('Viktor')
        }
    }
    return (
        <div>
            <button className="todos"
            disabled={!auth.status}
            onClick={props.onLoadTodos}
            >Todos</button>
            |
            <button
            onClick={props.onLoadAuth}
            >Auth</button>
            |
            <button 
              className="test"
              onClick={changePersonName}
              >
                {name === 'Viktor' ? 'Viktor' : 'Iryna'}
            </button>
        </div>
    );
}

export default Header;

























































// import React, { useContext } from 'react';
// import AuthContext from "../auth-context";

// const Header = (props) => {
//     const auth = useContext(AuthContext);
//     return (
//         <div>
//             {auth.status ? <button
//               onClick={props.onLoadTodos}
//             >Todos</button> : ''} | <button
//               onClick={props.onLoadAuth}
//             >Auth</button>
//             <hr />
//         </div>
//     );
// }

// export default Header;