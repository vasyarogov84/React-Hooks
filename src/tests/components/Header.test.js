import React from "react";
import Header from "../../../src/components/Header";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("Will test <Header /> component", () => {
  let wrapper = shallow(<Header />);
  test("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Should have 3 childs", () => {
    expect(wrapper.find("button").length).toBe(3);
    expect(wrapper.find(".todos").length).toBe(1);
  });
  test("Should show buttun text Todos", () => {
    expect(wrapper.find(".todos").text()).toBe("Todos");
    expect(wrapper.find("button").at(1).text()).toBe("Auth");
  });

  test("Should render context", () => {
    expect(wrapper.find(".test").text()).toBe("Viktor");
    wrapper.find(".test").simulate("click");
    expect(wrapper.find(".test").text()).toBe("Iryna");
    wrapper.find(".test").simulate("click");
    expect(wrapper.find(".test").text()).toBe("Viktor");
  });
});
