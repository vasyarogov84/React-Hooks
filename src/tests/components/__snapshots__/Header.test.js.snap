// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Will test <Header /> component should render correctly 1`] = `ShallowWrapper {}`;
